docs/lang/articles/deployment/tutorial.md

```diff
@@ -4,185 +4,215 @@ sidebar_position: 1
+The complete C++ source code is shown below, which is saved as `app.cpp` in the same directory as `app.py`.
 
-Integrate `TiRT` to your CMakeLists.txt:
+```c++
+#include <fstream>
+#include <taichi/cpp/taichi.hpp>
 
-```cmake
-# Find built taichi C-API library in `TAICHI_C_API_INSTALL_DIR`.
-find_library(taichi_c_api taichi_c_api HINTS ${TAICHI_C_API_INSTALL_DIR}/lib NO_CMAKE_FIND_ROOT_PATH)
-if (NOT EXISTS ${taichi_c_api})
-    message(FATAL_ERROR "Couldn't find C-API library in ${TAICHI_C_API_INSTALL_DIR}")
-endif()
-
-# Make sure your target is properly linked!
-set(TAICHI_TUTORIAL_DEMO_NAME "0_tutorial_kernel")
-message("-- Building ${TAICHI_TUTORIAL_DEMO_NAME}")
-add_executable(${TAICHI_TUTORIAL_DEMO_NAME} ${CMAKE_CURRENT_SOURCE_DIR}/app.cpp)
-target_include_directories(${TAICHI_TUTORIAL_DEMO_NAME} PUBLIC ${TAICHI_C_API_INSTALL_DIR}/include)
-target_link_libraries(${TAICHI_TUTORIAL_DEMO_NAME} ${taichi_c_api})
+void save_ppm(const float* pixels, uint32_t w, uint32_t h, const char* path) {
+  std::fstream f(path, std::ios::out | std::ios::trunc);
+  f << "P3\n" << w << ' ' << h << "\n255\n";
+  for (int i = 0; i < h; ++i) {
+    for (int j = 0; j < w; ++j) {
+      f << static_cast<uint32_t>(255.999 * pixels[(i * w + j) * 3 + 0]) << ' '
+        << static_cast<uint32_t>(255.999 * pixels[(i * w + j) * 3 + 1]) << ' '
+        << static_cast<uint32_t>(255.999 * pixels[(i * w + j) * 3 + 2]) << '\n';
+    }
+  }
+  f.flush();
+  f.close();
+}
+
+int main(int argc, const char** argv) {
+  ti::Runtime runtime(TI_ARCH_VULKAN);
+  ti::AotModule aot_module = runtime.load_aot_module("module.tcm");
+  ti::Kernel kernel_paint = aot_module.get_kernel("paint");
+
+  const uint32_t w = 512;
+  const uint32_t h = 512;
+  ti::NdArray<float> pixels = runtime.allocate_ndarray<float>({w, h}, {3}, true);
+
+  kernel_paint[0] = pixels;
+  kernel_paint.launch();
+  runtime.wait();
+
+  auto pixels_data = (const float*)pixels.map();
+  save_ppm(pixels_data, w, h, "result.ppm");
+  pixels.unmap();
+
+  return 0;
+}
 ```
 
-### 4. Run taichi kernels in your application
+### 3. Build project with CMake
 
-TiRT provides a fundamental C interface to help achieve optimal portability, however we also kindly provide a header-only C++ wrapper to save you from writing verbose C code. For simplicity purpose, we'll stick with the C++ wrapper in this tutorial.
+CMake is utilized to build the project. We introduce the file [`FindTaichi.cmake`](https://github.com/taichi-dev/taichi/blob/master/c_api/cmake/FindTaichi.cmake) to find Taichi C-API library, which is saved under the path `cmake/` in our working directory. The following shows the layout of our project.
 
-Calling Taichi in C++ as easy as what you'd imagine:
+```
+.
+├── cmake
+│   └── FindTaichi.cmake
+├── app.cpp
+├── app.py
+├── CMakeLists.txt
+└── module.tcm
+```
```

Comment:
Move this folder `ls` to the beginning of "Overview". Then we can briefly introduce what are these files and why we need any of them. Note that audiences might have different preferences respect to the sections. (E.g., an algorithm researcher might only want to learn how to write Taichi programs in Python, while a deployment engineer might only want to learn how to setup their C++ project).